tests/contrib/flask/test_request.py

```diff
@@ -0,0 +1,686 @@
+        self.assertEqual(len(spans), 8)
+
+        # Assert the order of the spans created
+        self.assertListEqual(
+            [
+                'flask.request',
+                'flask.try_trigger_before_first_request_functions',
+                'flask.preprocess_request',
+                'flask.dispatch_request',
+                'tests.contrib.flask.test_request.unicode',
+                'flask.process_response',
+                'flask.do_teardown_request',
+                'flask.do_teardown_appcontext',
+            ],
+            [s.name for s in spans],
+        )
+
+        # Assert span services
+        for span in spans:
+            self.assertEqual(span.service, 'flask')
+
+        # Root request span
+        req_span = spans[0]
+        self.assertEqual(req_span.service, 'flask')
+        self.assertEqual(req_span.name, 'flask.request')
+        self.assertEqual(req_span.resource, u'GET /üŋïĉóđē')
+        self.assertEqual(req_span.span_type, 'http')
+        self.assertEqual(req_span.error, 0)
+        self.assertIsNone(req_span.parent_id)
+
+        # Request tags
+        self.assertEqual(
+            set(['system.pid', 'flask.version', 'http.url', 'http.method',
+                 'flask.endpoint', 'flask.url_rule', 'http.status_code']),
+            set(req_span.meta.keys()),
+        )
+        self.assertEqual(req_span.get_tag('flask.endpoint'), 'unicode')
+        self.assertEqual(req_span.get_tag('flask.url_rule'), u'/üŋïĉóđē')
+        self.assertEqual(req_span.get_tag('http.method'), 'GET')
+        self.assertEqual(req_span.get_tag('http.url'), u'/üŋïĉóđē')
+        self.assertEqual(req_span.get_tag('http.status_code'), '200')
+
+        # Handler span
+        handler_span = spans[4]
+        self.assertEqual(handler_span.service, 'flask')
+        self.assertEqual(handler_span.name, 'tests.contrib.flask.test_request.unicode')
+        self.assertEqual(handler_span.resource, u'/üŋïĉóđē')
+        self.assertEqual(req_span.error, 0)
+
+    def test_request_404(self):
+        """
+        When making a request
+            When the requested endpoint was not found
+                We create the expected spans
+        """
+        res = self.client.get('/not-found')
+        self.assertEqual(res.status_code, 404)
+
+        spans = self.get_spans()
+        self.assertEqual(len(spans), 9)
+
+        # Assert the order of the spans created
+        self.assertListEqual(
+            [
+                'flask.request',
+                'flask.try_trigger_before_first_request_functions',
+                'flask.preprocess_request',
+                'flask.dispatch_request',
+                'flask.handle_user_exception',
+                'flask.handle_http_exception',
+                'flask.process_response',
+                'flask.do_teardown_request',
+                'flask.do_teardown_appcontext',
+            ],
+            [s.name for s in spans],
+        )
+
+        # Assert span serices
```

Comment:
```suggestion
        # Assert span services
```

Reply:
lol, 1) I can't spell/type to save my life, and 2) I copy/pasta way too much 😆